### ISSUE TITLE
Cannot get comments to work...

### ISSUE THREAD

**Issue by cwerner** [mmistakes/minimal-mistakes#1594]
- [x] This is a question about using the theme.
- [ ] This is a feature request or enhancement.
- [ ] I believe this to be a bug with the theme.
  - [x] I am using the latest version of the theme.
  - [x] I have updated all gems with `bundle update`.
  - [x] I have tested locally with `bundle exec jekyll build`.

---

## Environment informations

- **Minimal Mistakes version:** 
- **Jekyll version:**
- **GitHub Pages hosted:** yes 
- **Operating system:** 

---

## Expected behavior

Get comments (statesman_v2)

## Steps to reproduce the behavior

https://github.com/cwerner/chrwerner.github.io

Hi.

This may very well be me butchering the theme/ files out of desperation, but i cannot get the comments to show for the live of me. I started out with jekyll now and migrated to mmistakes. Finally, I tried to switch to the gem install setup.
Not knowing my way around ruby maybe I nuked a file I needed?

Could any kind soul have a look at the repo? The site lives at https://www.christianwerner.net

Thanks!

`$ bundle exec jekyll build --trace --verbose
  Logging at level: debug
Configuration file: /Users/cwerner/Dropbox/development/chrwerner.github.io/_config.yml
      GitHub Pages: github-pages v179
      GitHub Pages: jekyll v3.6.2
         Requiring: jekyll-seo-tag
         Requiring: jekyll-github-metadata
Dotenv not found. Skipping 
         Requiring: jekyll-paginate
         Requiring: jekyll-sitemap
         Requiring: jekyll-gist
         Requiring: jekyll-feed
         Requiring: jemoji
         Requiring: jekyll-coffeescript
         Requiring: jekyll-commonmark-ghpages
         Requiring: jekyll-github-metadata
         Requiring: jekyll-relative-links
         Requiring: jekyll-optional-front-matter
         Requiring: jekyll-readme-index
         Requiring: jekyll-default-layout
         Requiring: jekyll-titles-from-headings
         Requiring: jekyll-remote-theme
         Requiring: kramdown
   GitHub Metadata: Initializing...
            Source: /Users/cwerner/Dropbox/development/chrwerner.github.io
       Destination: /Users/cwerner/Dropbox/development/chrwerner.github.io/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
      Remote Theme: Using theme mmistakes/minimal-mistakes
      Remote Theme: Downloading https://codeload.github.com/mmistakes/minimal-mistakes/zip/master to /var/folders/7k/gqvwq21j2gxd0px09mq154j00000gn/T/jekyll-remote-theme-20180320-9240-664vvm.zip
ETHON: Libcurl initialized
  Trying 192.30.253.120...
TCP_NODELAY set
Connected to codeload.github.com (192.30.253.120) port 443 (#0)
ALPN, offering http/1.1
Cipher selection: ALL:!EXPORT:!EXPORT40:!EXPORT56:!aNULL:!LOW:!RC4:@STRENGTH
successfully set certificate verify locations:
  CAfile: /etc/ssl/cert.pem
  CApath: none
TLSv1.2 (OUT), TLS handshake, Client hello (1):
? ?4?}???@
          ??Z?C?30??S*??? ??????0?,?(?$??
??kj98????????2?.?*?&???=5???/?+?'?#??	??g@32??ED?1?-?)?%???</?A??
?Mcodeload.github.com

:8

	
&$??????3t
               http/1.1?TLSv1.2 (IN), TLS handshake, Server hello (2):
h???24????:??hcn??z:?£?'k?? ? !VA~e???UD??9?G???7?ɍ4?;???/ ?

                                                               http/1.1TLSv1.2 (IN), TLS handshake, Certificate (11):

???Ϭa?Æo*?H??0
0p1
   0	UUS10U

DigiCert Inc10U
200417120000Z0h1ww.digicert.com1/0-U&DigiCert SHA2 High Assurance Server CA0
                0	UUS10U
San Francisco10U

GitHub, Inc.10U

?0?     *?H??  *.github.com0?"0
????0I-??

        ??px~k???U6?i?_???NR?HCT?
?Y?Yj?F?vN?t???K?????\??<?? ???.?⬇G??C??????!?G?ADmӥ-k??.??t?˩;?K????t?姮??,?l?????????|2?<??F?????g+?ʏp*\?j?JCa}Z
??kφ??>?]މ?e???0??0U#0?Qh???u<??edb??Yr;0U?aR??x??`?`
@?? ?07$??lj?0#U0?
                  *.github.com?
github.com0U??0U%+0uUn0l04?2?0?.http://crl3.digicert.com/sha2-ha-server-g5.crl04?2?0?.http://crl4.digicert.com/sha2-ha-server-g5.crl0LU E0C07	`?H??l0*0+https://www.digicert.com/CPSg?
                        0?+w0u0+0?http://ocsp.digicert.com0+0?Ahttp://cacerts.digicert.com/DigiCertSHA2HighAssuranceServerCA.crt0
                                                 U?00??
+?y?????u??	??X????gp
?Y??>KF0D ??ѵSGV?)>??A]4??????27??w?B ?*(??g:?4>??V??H?L??B?vV?/??????D?>?Fv???\?U։??Y???G0E $?i??(??-|????Eʊǂ?]?? ,T?ھ!?%???e?ܠ?b???%?K-~I?1W??V?.?Wu?K??u?`??Bi??f?~_?r؃?{?z???Y??@.F0D r??Ho3D??6f?p?ܹD???&?4x&\??? F??iՔn;?Z [ՙđ??_?Ͷ??Xߢ$?v??߼?q???#???{G8W?
?8?H? 4??(????R0E!?L?	]????E=6?WZ??Қ9
        *?H??  ŵD[.??? /"???N?&[u0
?|?/IJ?g??"=年?t?????9K?̏???????,???3ǰV??i?{X?E.	????v??2˶?&?/_??i?Xh?\??vp	?V???H?E??|*BD?5?s??Vs+?K?X?a????_FD¥???V6߬??n?4Y???????!}y&(R??+???Du)?,??????ϵ????*???OY*?H???3?1n?x(?˕)??0??0??????\??m?+B?]?0
0l1
   0	UUS10U

DigiCert Inc10U
281022120000Z0p1ww.digicert.com1+0)U"DigiCert High Assurance EV Root CA0
                0	UUS10U

DigiCert Inc10U
?0?     *?H??  www.digicert.com1/0-U&DigiCert SHA2 High Assurance Server CA0?"0
???/?$?m_??
d?}"&e?B@??ܟ?v>?0U??O?Z?UV??*?И?
                                @???;r?????ȸC:?

                                                ?Օ?@???M?_W??Ph?????????ܤ???-ƭ??????^DX*7??5?
x?{?^q?J?????#M??u?????D5eS%9?(Ļ(̚?+#
?cth
7??RH?9Z???]?ݠ ?!??&oJ!A??m??H/???h?S/^???I0?E0U0?0U??0U%+0+(0&0+0?http://ocsp.digicert.com0KUD0B0@?>?<?:http://crl4.digicert.com/DigiCertHighAssuranceEVRootCA.crl0=U 60402U 0*0+https://www.digicert.com/CPS0UQh???u<??edb??Yr;0U#0??>?i??GԘ?cd+?0    *?H??
?????m?\?h?J???Q/?kD??c??nl?????q?[?4N??y?)?-?j?? ?y??G????Yq}??k?YX=??1%\8??柂??[?1N?x????I??'??r?>??A
                       ??6??nGI^?H|???I??&B@?֒?d
WT???^k??Ā?r?V????0?0?
                      N?W?$??+??u??-?}y'??
                                           ꪅ?? (AYC(ҁ???{;w?@b?AE?>??3?g?a.r??i?W@p?ɴTLSv1.2 (IN), TLS handshake, Server key exchange (12):

IA)??
??fj܀?ޖ????????????'??I?j???!???߲t??
??.??-3??}C"?P??kQ^c?C?P??ɶ;???뾆??&?~??4?*??????Y???
                            P???D??????v#??x?oA?5p?&?lbLJ|z?^R?
???5+$???\@4S?l??ή?z????b?,?|?;?Rk?=??N???LU?.6?????? ??1M???nY?%?>ˮ?G0Yz??p$?d2? ??P?IVfޱNU8Lu???
                  O???\x?j?dS?? ?~??J$???ITLSv1.2 (IN), TLS handshake, Server finished (14):
TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
BA???????Tel?o/!H???????LN=j?83?
                                ts?u??BA???#??:c??p????2{TLSv1.2 (OUT), TLS change cipher, Client hello (1):
TLSv1.2 (OUT), TLS handshake, Finished (20):

K?®??<??7TLSv1.2 (IN), TLS change cipher, Client hello (1):
TLSv1.2 (IN), TLS handshake, Finished (20):

?	??1;??C??RSSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
ALPN, server accepted to use http/1.1
Server certificate:
 subject: C=US; ST=California; L=San Francisco; O=GitHub, Inc.; CN=*.github.com
 start date: Jan 18 00:00:00 2017 GMT
 expire date: Apr 17 12:00:00 2020 GMT
 subjectAltName: host "codeload.github.com" matched cert's "*.github.com"
 issuer: C=US; O=DigiCert Inc; OU=www.digicert.com; CN=DigiCert SHA2 High Assurance Server CA
 SSL certificate verify ok.
GET /mmistakes/minimal-mistakes/zip/master HTTP/1.1
Host: codeload.github.com
Accept: */*
User-Agent: Typhoeus - https://github.com/typhoeus/typhoeus
user_agent: Jekyll Remote Theme/0.2.3 (+https://github.com/benbalter/jekyll-remote-theme)

HTTP/1.1 200 OK
Content-Length: 25895749
Access-Control-Allow-Origin: https://render.githubusercontent.com
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Strict-Transport-Security: max-age=31536000
Vary: Authorization,Accept-Encoding
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
ETag: "fbb7356d74995fb874258cfc37e74b31bd1b318b"
Content-Type: application/zip
Content-Disposition: attachment; filename=minimal-mistakes-master.zip
X-Geo-Block-List: 
Date: Mon, 19 Mar 2018 23:10:11 GMT
X-GitHub-Request-Id: DDDE:4590:1F8B0B:34D111:5AB04353

Connection #0 to host codeload.github.com left intact
ETHON: performed EASY effective_url=https://codeload.github.com/mmistakes/minimal-mistakes/zip/master response_code=200 return_code=ok total_time=20.026652
      Remote Theme: Unzipping /var/folders/7k/gqvwq21j2gxd0px09mq154j00000gn/T/jekyll-remote-theme-20180320-9240-664vvm.zip to /private/var/folders/7k/gqvwq21j2gxd0px09mq154j00000gn/T/jekyll-remote-theme-20180320-9240-1ua5kzi
       EntryFilter: excluded /test
       EntryFilter: excluded /CHANGELOG.md
       EntryFilter: excluded /docs
       EntryFilter: excluded /README.md
       EntryFilter: excluded /Rakefile
       EntryFilter: excluded /package.json
       EntryFilter: excluded /Gemfile
       EntryFilter: excluded /Gemfile.lock
       EntryFilter: excluded /LICENSE.txt
           Reading: _posts/2018-1-2-Hello-World.md
           Reading: _posts/2012-01-02-layout-comments-disabled.md
           Reading: _posts/2012-01-02-layout-comments.md
        Generating: JekyllOptionalFrontMatter::Generator finished in 4.4e-05 seconds.
        Generating: JekyllReadmeIndex::Generator finished in 8.9e-05 seconds.
        Generating: JekyllFeed::Generator finished in 0.002189 seconds.
        Generating: JekyllRelativeLinks::Generator finished in 0.004802 seconds.
        Generating: JekyllDefaultLayout::Generator finished in 0.004818 seconds.
        Generating: Jekyll::Paginate::Pagination finished in 0.001068 seconds.
        Generating: JekyllTitlesFromHeadings::Generator finished in 0.000222 seconds.
        Generating: Jekyll::JekyllSitemap finished in 0.001535 seconds.
         Rendering: _posts/2012-01-02-layout-comments-disabled.md
  Pre-Render Hooks: _posts/2012-01-02-layout-comments-disabled.md
  Rendering Liquid: _posts/2012-01-02-layout-comments-disabled.md
  Rendering Markup: _posts/2012-01-02-layout-comments-disabled.md
         Requiring: kramdown
  Rendering Layout: _posts/2012-01-02-layout-comments-disabled.md
         Rendering: _posts/2012-01-02-layout-comments-disabled.md
  Pre-Render Hooks: _posts/2012-01-02-layout-comments-disabled.md
  Rendering Liquid: _posts/2012-01-02-layout-comments-disabled.md
  Rendering Markup: _posts/2012-01-02-layout-comments-disabled.md
         Rendering: _posts/2018-1-2-Hello-World.md
  Pre-Render Hooks: _posts/2018-1-2-Hello-World.md
  Rendering Liquid: _posts/2018-1-2-Hello-World.md
  Rendering Markup: _posts/2018-1-2-Hello-World.md
         Rendering: _posts/2012-01-02-layout-comments.md
  Pre-Render Hooks: _posts/2012-01-02-layout-comments.md
  Rendering Liquid: _posts/2012-01-02-layout-comments.md
  Rendering Markup: _posts/2012-01-02-layout-comments.md
         Rendering: _posts/2012-01-02-layout-comments.md
  Pre-Render Hooks: _posts/2012-01-02-layout-comments.md
  Rendering Liquid: _posts/2012-01-02-layout-comments.md
  Rendering Markup: _posts/2012-01-02-layout-comments.md
  Rendering Layout: _posts/2012-01-02-layout-comments.md
         Rendering: _posts/2018-1-2-Hello-World.md
  Pre-Render Hooks: _posts/2018-1-2-Hello-World.md
  Rendering Liquid: _posts/2018-1-2-Hello-World.md
  Rendering Markup: _posts/2018-1-2-Hello-World.md
  Rendering Layout: _posts/2018-1-2-Hello-World.md
         Rendering: _pages/about.md
  Pre-Render Hooks: _pages/about.md
  Rendering Liquid: _pages/about.md
  Rendering Markup: _pages/about.md
  Rendering Layout: _pages/about.md
         Rendering: _pages/archive-layout-with-content.md
  Pre-Render Hooks: _pages/archive-layout-with-content.md
  Rendering Liquid: _pages/archive-layout-with-content.md
  Rendering Markup: _pages/archive-layout-with-content.md
  Rendering Layout: _pages/archive-layout-with-content.md
         Rendering: _pages/category-archive.html
  Pre-Render Hooks: _pages/category-archive.html
  Rendering Liquid: _pages/category-archive.html
  Rendering Markup: _pages/category-archive.html
  Rendering Layout: _pages/category-archive.html
         Rendering: _pages/home.md
  Pre-Render Hooks: _pages/home.md
  Rendering Liquid: _pages/home.md
  Rendering Markup: _pages/home.md
  Rendering Layout: _pages/home.md
         Rendering: index.html
  Pre-Render Hooks: index.html
  Rendering Liquid: index.html
  Rendering Markup: index.html
  Rendering Layout: index.html
         Rendering: _pages/portfolio-archive.html
  Pre-Render Hooks: _pages/portfolio-archive.html
  Rendering Liquid: _pages/portfolio-archive.html
  Rendering Markup: _pages/portfolio-archive.html
  Rendering Layout: _pages/portfolio-archive.html
         Rendering: _pages/projects.md
  Pre-Render Hooks: _pages/projects.md
  Rendering Liquid: _pages/projects.md
  Rendering Markup: _pages/projects.md
  Rendering Layout: _pages/projects.md
         Rendering: _pages/sitemap.md
  Pre-Render Hooks: _pages/sitemap.md
  Rendering Liquid: _pages/sitemap.md
  Rendering Markup: _pages/sitemap.md
  Rendering Layout: _pages/sitemap.md
         Rendering: _pages/tag-archive.html
  Pre-Render Hooks: _pages/tag-archive.html
  Rendering Liquid: _pages/tag-archive.html
  Rendering Markup: _pages/tag-archive.html
  Rendering Layout: _pages/tag-archive.html
         Rendering: _pages/terms.md
  Pre-Render Hooks: _pages/terms.md
  Rendering Liquid: _pages/terms.md
  Rendering Markup: _pages/terms.md
  Rendering Layout: _pages/terms.md
         Rendering: assets/css/main.scss
  Pre-Render Hooks: assets/css/main.scss
  Rendering Liquid: assets/css/main.scss
  Rendering Markup: assets/css/main.scss
         Rendering: assets/js/lunr/lunr-en.js
  Pre-Render Hooks: assets/js/lunr/lunr-en.js
  Rendering Liquid: assets/js/lunr/lunr-en.js
  Rendering Markup: assets/js/lunr/lunr-en.js
  Rendering Layout: assets/js/lunr/lunr-en.js
         Rendering: assets/js/lunr/lunr-gr.js
  Pre-Render Hooks: assets/js/lunr/lunr-gr.js
  Rendering Liquid: assets/js/lunr/lunr-gr.js
  Rendering Markup: assets/js/lunr/lunr-gr.js
  Rendering Layout: assets/js/lunr/lunr-gr.js
         Rendering: assets/js/lunr/lunr-store.js
  Pre-Render Hooks: assets/js/lunr/lunr-store.js
  Rendering Liquid: assets/js/lunr/lunr-store.js
  Rendering Markup: assets/js/lunr/lunr-store.js
  Rendering Layout: assets/js/lunr/lunr-store.js
         Rendering: feed.xml
  Pre-Render Hooks: feed.xml
  Rendering Liquid: feed.xml
  Rendering Markup: feed.xml
  Rendering Layout: feed.xml
         Rendering: sitemap.xml
  Pre-Render Hooks: sitemap.xml
  Rendering Liquid: sitemap.xml
  Rendering Markup: sitemap.xml
  Rendering Layout: sitemap.xml
         Rendering: robots.txt
  Pre-Render Hooks: robots.txt
  Rendering Liquid: robots.txt
  Rendering Markup: robots.txt
  Rendering Layout: robots.txt
                    done in 24.588 seconds.
 Auto-regeneration: disabled. Use --watch to enable.
      Remote Theme: Cleaning up /private/var/folders/7k/gqvwq21j2gxd0px09mq154j00000gn/T/jekyll-remote-theme-20180320-9240-1ua5kzi
`

**Comment by mmistakes** [comment]
As noted in the documentation and several times previously in related issues you need to set Jekyll's environment variable to `production`. There is a condition in the theme that will only show the comments section when it is `production`, you've set a `comments.provider` in _config.yml, and `comments: true` in the post's front matter.

> **Note:** Comments are disabled by default in `development`. To enable when testing/building locally be sure to set `JEKYLL_ENV=production` to [force the environment](http://jekyllrb.com/docs/configuration/#specifying-a-jekyll-environment-at-build-time) to `production`.
>
> ref: https://mmistakes.github.io/minimal-mistakes/docs/configuration/#comments

Related issues: #1562, #1544

**Comment by mmistakes** [comment]
Also comments appear to be working fine as far as I can tell. You have the comment form on your posts.

https://www.christianwerner.net/tech/Hello-World/

You don't have any comments created by Staticman yet... (you'd know because you'd have files in `_data/comments/`).

I just submitted a comment on that post and it went through.

![image](https://user-images.githubusercontent.com/1376749/37663698-2767c6be-2c30-11e8-85cc-0d210de5ff2b.png)

Closing as this seems to be a non-issue.

**Comment by cwerner** [comment]
Thanks for looking into this.

I knew about the production flag, and to be honest might have accidentally fixed comments in my fooling around relentlessly…
I think I had a half working/ corrupt site since I migrated it from a default theme and messed something up. Also, I removed the gem-included folders… not sure if this caused something…

Again, thank you and keep up the great work. Really nice theme you created there…

Christian
